sys/include/net/ng_llhdr.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net
+ * @{
+ *
+ * @file
+ * @brief       Generic link layer header
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef LLHDR_H_
+#define LLHDR_H_
+
+#include <string.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Generic link layer header
+ *
+ * The actual addresses are put into the memory following this datastructure.
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t addr_len;           /**< link layer address length in byte */
```

Comment:
See e. g. test case TD_6LoWPAN_FORMAT_07 in the [6LoWPAN Plugtests](https://github.com/authmillenon/projects/blob/6LoWPAN_PLUGTEST/6lpplugtest/README.md#test-descriptions-by-etsi). There is a way (or let's say 2 ways) to translate a 16-bit address to an EUI-64, but this is part of the 6LoWPAN standard (i. e. other adaptions might differ). But even 6LoWPAN presents 2 ways to do this. Our current implementation translates iirc the 16-bit address using the PAN ID as it is needed for [the older HC1](https://tools.ietf.org/html/rfc4944#section-6), but we only support IPHC which [doesn't need the PAN ID anymore](https://tools.ietf.org/html/rfc6282#section-3.2.2), so the address gets retranslated into that format again. To prevent such shenanigans I originally introduced `src_len` and `dest_len` in the original `netapi`.